contracts/src/components/escrow/escrow.cairo

```diff
@@ -0,0 +1,97 @@
+#[starknet::component]
+pub mod EscrowComponent {
+    use openzeppelin_token::erc20::interface::{IERC20Dispatcher, IERC20DispatcherTrait};
+    use starknet::ContractAddress;
+    use starknet::storage::Map;
+    use zkramp::components::escrow::interface;
+
+    //
+    // Storage
+    //
+
+    #[storage]
+    struct Storage {
+        // (owner, token) -> amount
+        deposits: Map::<(ContractAddress, ContractAddress), u256>,
+        // token -> escrow address
+        escrow_contract_addresses: Map::<ContractAddress, ContractAddress>,
+    }
+
+    //
+    // Errors
+    //
+
+    pub mod Errors {
+        pub const PROOF_OF_DEPOSIT_FAILED: felt252 = 'Proof of deposit failed';
+        pub const INSUFFICIENT_BALANCE: felt252 = 'Insufficient deposit balance';
+    }
+
+    //
+    // Escrow impl
+    //
+
+    #[embeddable_as(RegistryImpl)]
+    impl Escrow<
+        TContractState, +HasComponent<TContractState>, +Drop<TContractState>,
+    > of interface::IEscrow<ComponentState<TContractState>> {
+        fn lock_from(
+            ref self: ComponentState<TContractState>,
+            from: ContractAddress,
+            token: ContractAddress,
+            amount: u256
+        ) {
+            let erc20_dispatcher = IERC20Dispatcher { contract_address: token };
+
+            let balance = erc20_dispatcher.balance_of(from);
+
+            assert(balance >= amount, Errors::INSUFFICIENT_BALANCE);
```

Comment:
This check is useless, the transfer will fail anyway if the balance is insufficient